docs/advanced_topics/accessibility_considerations.rst

```diff
@@ -0,0 +1,160 @@
+- For content that is displayed in a specific section of the page, limit heading levels to those directly below the section’s main heading.
+
+If managing headings via StreamField, make sure to apply the same restrictions there.
+
+Bold and italic formatting in rich text
+---------------------------------------
+
+By default, Wagtail stores its bold formatting as a ``b`` tag, and italic as ``i`` (`#4665 <https://github.com/wagtail/wagtail/issues/4665>`_). While those tags don’t necessarily always have correct semantics (``strong`` and ``em`` are more ubiquitous), there isn’t much consequence for screen reader users, as by default screen readers do not announce content differently based on emphasis.
+
+If this is a concern to you, you can change which tags are used when saving content with :ref:`rich text format converters <rich_text_format_converters>`. In the future, :ref:`rich text rewrite handlers <rich_text_rewrite_handlers>` should also support this being done without altering the storage format (`#4223 <https://github.com/wagtail/wagtail/issues/4223>`_).
+
+TableBlock
+----------
+
+The :doc:`/reference/contrib/table_block` default implementation makes it too easy for end users to miss they need either row or column headers (`#5989 <https://github.com/wagtail/wagtail/issues/5989>`_). Make sure to always have either row headers or column headers set.
+Also treat its Caption field as mandatory, so screen reader users navigating the site’s tables know where they are.
+
+----
+
+.. _accessibility_in_templates:
+
+Accessibility in templates
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Here are common gotchas to be aware of to make the site’s templates as accessible as possible,
+
+Alt text in templates
+---------------------
+
+See the :ref:`content modelling <content_modeling>` section above. Additionally, make sure to :ref:`customise images’ alt text <image_tag_alt>`, either setting it to the relevant field, or to an empty string for decorative images, or images where the alt text would be a repeat of other content.
+Even when your images have alt text coming directly from the image model, you still need to decide whether there should be alt text for the particular context the image is used in. For example, avoid alt text in listings where the alt text just repeats the listing items’ title.
+
+Empty heading tags
+------------------
+
+In both rich text and custom StreamField blocks, it’s sometimes easy for editors to create a heading block but not add any content to it. If this is a problem for your site,
+
+- Add validation rules to those fields, making sure the page can’t be saved with the empty headings.
+- Consider adding similar validation rules for rich text fields.
+
+Additionally, you can hide empty heading blocks with CSS:
+
+.. code-block:: css
+
+    h1:empty, h2:empty, h3:empty, h4:empty, h5:empty, h6:empty {
```

Comment:
Good question – this has been discussed before in #4431, and I don’t know what to recommend. I don’t think there are accessibility concerns with empty paragraphs. I need to check for list items.

Reply:
Is this a candidate to be handled by the CMS? If empty headings and empty paragraphs are bad. Why are they not filtered out? Of raise a validation error on save? Why would we promote hacks?

Reply:
Empty paragraphs aren’t bad for accessibility so let’s leave this to #4431. Empty headings are bad, so I’m going to add documentation on how to make sure StreamField heading blocks are required. For rich text, I have opened #6526.

I think promoting the CSS trick is appropriate in this case, because I assume it’s an issue that affects most / all Wagtail sites right now, and there simply is no other ready-to-go solution to it.